sdk/webpubsub/Azure.Messaging.WebPubSub.Client/src/EventSource/WebPubSubClientEventSource.cs

```diff
@@ -0,0 +1,155 @@
+
+        [Event(1, Level = EventLevel.Verbose, Message = "Client starting.")]
+        public virtual void ClientStarting()
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(ClientStartingId);
+            }
+        }
+
+        [Event(2, Level = EventLevel.Verbose, Message = "The client state transient from the '{0}' state to the '{1}' state.")]
+        public virtual void ClientStateChanged(string newState, string currentState)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(ClientStateChangedId, currentState, newState);
+            }
+        }
+
+        [Event(3, Level = EventLevel.Verbose, Message = "A new WebSocket connection start to connect with subprotocol {0}.")]
+        public virtual void WebSocketConnecting(string subprotocol)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(WebSocketConnectingId, subprotocol);
+            }
+        }
+
+        [Event(4, Level = EventLevel.Verbose, Message = "WebSocket connection closed.")]
+        public virtual void WebSocketClosed()
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(WebSocketClosedId);
+            }
+        }
+
+        [Event(5, Level = EventLevel.Warning, Message = "An exception occurred while process message from the service. Error Message: {0}")]
+        public virtual void FailedToProcessMessage(string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(FailedToProcessMessageId, errorMessage);
+            }
+        }
+
+        [Event(6, Level = EventLevel.Informational, Message = "An exception occurred while receiving bytes. Error Message: {0}")]
+        public virtual void FailedToReceiveBytes(string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(FailedToReceiveBytesId, errorMessage);
+            }
+        }
+
+        [Event(7, Level = EventLevel.Warning, Message = "The client failed to transition from the '{0}' state to the '{1}' state because it was actually in the '{2}' state.")]
+        public virtual void FailedToChangeClientState(string expectedState, string newState, string currentState)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(FailedToChangeClientStateId, expectedState, newState, currentState);
+            }
+        }
+
+        [Event(8, Level = EventLevel.Warning, Message = "Stop try to recover the connection with connection ID: {0} cause of {1}.")]
+        public virtual void StopRecovery(string connectionId, string reason)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(StopRecoveryId, connectionId, reason);
+            }
+        }
+
+        [Event(9, Level = EventLevel.Informational, Message = "An attempt to recover connection with connection ID: {0} failed, will retry later. Error Message: {1}")]
```

Comment:
```suggestion
        [Event(9, Level = EventLevel.Informational, Message = "An attempt to recover connection with connection ID: {0} is failed, will retry later. Error Message: {1}")]
```

Reply:
The original version is proper grammar; the suggestion is not.